Lib/test/test_tracemalloc.py

```diff
@@ -246,6 +246,35 @@ def test_clear_traces(self):
         traceback2 = tracemalloc.get_object_traceback(obj)
         self.assertIsNone(traceback2)
 
+    def test_reset_peak(self):
```

Comment:
I expected a test similar to https://mail.python.org/archives/list/python-ideas@python.org/thread/QDWI37A4TJXOYUKULGPY2GKD7IG2JNDC/ use case:

* Allocate large memory block (ex: 10 MB)
* Allocate small memory block (ex: 1 MB)
* Check that peak is greater than or equal to the large block
* Free the large block
* Reset peak
* Check that peak is smaller than the large block

I would prefer assertIn() which is looks fragile depending on the exact amount of memory allocated internally by Python.

Reply:
Hm, I'm not sure I understand. I think the test is already doing basically that, except being slightly more precise about the sizes:

1. line 263 checks the peak size is what's expected
2. line 270 checks the peak size is smaller (~equal to the current size)

Just so I understand, are you saying you would be happier with a smaller/simpler test?

> I would prefer assertIn() which is looks fragile depending on the exact amount of memory allocated internally by Python.

Sorry, I'm a bit confused by this: did you mean "I would prefer not using `assertIn()`?

I used `range` + `assertIn` as "all in one" version of the pairs of `assertGreaterEqual(x, expected)` & `assertLessEqual(x - expected, max_error)` calls in the `test_get_traced_memory` test, and I think it's equivalent (except for being excluding the `expected + max_error` endpoint). Are you concerned that the `max_error` approach is too fragile? I could increase `max_error` to 10k or more?

Reply:
Yeah, max_error looks fragile. The test doesn't have to check for exact number of bytes, only that new peak is greater or less than the previous peak.

Here is a test based on your example and test_get_traced_memory():

```
    def test_reset_peak(self):
        # Python allocates some internals objects, so the test must tolerate
        # a small difference between the expected size and the real usage
        tracemalloc.clear_traces()

        # Example: allocate a large piece of memory, temporarily
        large_sum = sum(list(range(100000)))
        size1, peak1 = tracemalloc.get_traced_memory()

        # reset_peak() resets peak to traced memory: peak2 < peak1
        tracemalloc.reset_peak()
        size2, peak2 = tracemalloc.get_traced_memory()
        self.assertGreaterEqual(peak2, size2)
        self.assertLess(peak2, peak1)

        # check that peak continue to be updated if new memory is allocated:
        # peak3 > peak2
        obj_size = 1024 * 1024
        obj, obj_traceback = allocate_bytes(obj_size)
        size3, peak3 = tracemalloc.get_traced_memory()
        self.assertGreaterEqual(peak3, size3)
        self.assertGreater(peak3, peak2)
        self.assertGreaterEqual(peak3 - peak2, obj_size)
```

I use `assertGreaterEqual(peak, size)` since it's ok if the peak is greater than the traced size. Python can allocates memory in our back, there are tons of internal caches.

`self.assertGreaterEqual(peak3 - peak2, obj_size)` looks safe: we ensure that obj_size contributes to the new peak value.

Reply:
Ok, I've gone with your suggested test.

Would it make sense to update `test_get_traced_memory` to avoid using the `max_error` technique, at some point too? (Not in this PR.)